libqtile/backend/base/zmanager.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2025 elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import IntEnum
+from functools import wraps
+
+from libqtile.backend.base.window import _Window
+
+
+class LayerGroup(IntEnum):
+    BACKGROUND = 0
+    BOTTOM = 1
+    KEEP_BELOW = 2
+    LAYOUT = 3
+    KEEP_ABOVE = 4
+    MAX = 5
+    FULLSCREEN = 6
+    BRING_TO_FRONT = 7
+    TOP = 8
+    OVERLAY = 9
+    SYSTEM = 10
```

Comment:
Do you need something here that's OVERRIDE_REDIRECT i.e. "don't mess with the stacking of this window"? But I guess it's not really "don't mess with it" but "keep it relative to its parent window"?